apps/src/pythonlab/aiTutorHelper.ts

```diff
@@ -6,7 +6,7 @@ import PythonValidationTracker from './progress/PythonValidationTracker';
 
 // Return additional context for AiTutor2.
 const getAiTutor2Context = (
-  source: MultiFileSource,
+  source: MultiFileSource | undefined,
```

Comment:
this was a safe change to make because we check for the existence of `source` as the first step.